activerecord/lib/active_record/relation.rb

```diff
@@ -219,12 +219,23 @@ def find_or_create_by(attributes, &block)
     # Like <tt>find_or_create_by</tt>, but calls <tt>create!</tt> so an exception
     # is raised if the created record is invalid.
     def find_or_create_by!(attributes, &block)
-      find_by(attributes) || create!(attributes, &block)
```

Comment:
yeah, may be better...just wasn't sure if it was really appropriate there...can't really think of where you'd want to `find_by` with a block outside of the context of one of these two methods